ggircs-app/app/components/FuelManagement/CarbonTaxActFuelType.tsx

```diff
@@ -0,0 +1,202 @@
+import { Card, Col, Row, Table } from "react-bootstrap";
+import React from "react";
+import { GraphQLTaggedNode, useFragment, graphql } from "react-relay";
+import { CarbonTaxActFuelType_query$key } from "__generated__/CarbonTaxActFuelType_query.graphql";
+import { FuelSelectionComponent } from "./FuelSelectionComponent";
+import { FuelChargeRow } from "./FuelChargeRow";
+import CreateEditFuelChargeRow from "./CreateEditFuelChargeRow";
+import { useRouter } from "next/router";
+import Button from "@button-inc/bcgov-theme/Button";
+
+interface Props {
+  query: CarbonTaxActFuelType_query$key;
+  pageQuery: GraphQLTaggedNode;
+}
+
+export const CarbonTaxActFuelType: React.FC<Props> = ({ query, pageQuery }) => {
+  const { carbonTaxActFuelType, allCarbonTaxActFuelTypes } = useFragment(
+    graphql`
+      fragment CarbonTaxActFuelType_query on Query
+      @argumentDefinitions(carbonTaxActFuelTypeId: { type: "ID!" }) {
+        carbonTaxActFuelType(id: $carbonTaxActFuelTypeId) {
+          id
+          rowId
+          carbonTaxFuelType
+          ctaRateUnits
+          fuelChargesByCarbonTaxActFuelTypeId(
+            first: 1000
+            orderBy: START_DATE_ASC
+          )
+            @connection(
+              key: "CarbonTaxActFuelType_fuelChargesByCarbonTaxActFuelTypeId"
+            ) {
+            __id
+            edges {
+              node {
+                id
+                startDate
+                endDate
+                ...FuelChargeRow_fuelCharge
+              }
+            }
+          }
+        }
+        allCarbonTaxActFuelTypes: query {
+          allCarbonTaxActFuelTypes {
+            edges {
+              node {
+                id
+                carbonTaxFuelType
+              }
+            }
+          }
+        }
+      }
+    `,
+    query
+  );
+
+  const router = useRouter();
+
+  const validateRatePeriod = (date: string, id?: string) => {
+    const dateToValidate = new Date(date);
+    const invalidDate =
+      carbonTaxActFuelType.fuelChargesByCarbonTaxActFuelTypeId.edges
+        .filter((edge) => edge.node.id !== id)
+        .some(
+          (edge) =>
+            new Date(edge.node.startDate) <= dateToValidate &&
+            new Date(edge.node.endDate) >= dateToValidate
+        );
+    if (invalidDate) return false;
+
+    return true;
+  };
+
+  const fromIndex =
+    router.query.showAll === "true"
+      ? 0
+      : carbonTaxActFuelType?.fuelChargesByCarbonTaxActFuelTypeId?.edges
+          ?.length - 3;
```

Comment:
why `3` ? Not sure what's going on here

Reply:
I figure most of the time there will be rows for an upcoming charge, a current charge and the previous year's charge at the end of the list, so I set the default number of rows to show at 3 until the 'show all' button is clicked. I'll add a comment.

Reply:
A variable name would be better than a comment :)